pkg/plugins/golang/v2/scaffolds/internal/templates/gomod.go

```diff
@@ -50,5 +50,6 @@ go 1.13
 
 require (
 	sigs.k8s.io/controller-runtime {{ .ControllerRuntimeVersion }}
+	golang.org/x/sys v0.0.0-20220330033206-e17cdc41300f // indirect
```

Comment:
Why are you adding it here?
IMO it shows that we need to upgrade all plugins to use go 1.18

Reply:
that's probably the result of `go mod tidy`

Reply:
Because there's a bug with the version that the `go mod tidy` imports(an older one due to old dependencies), which this version solves.

Here, I manually override the old dependency with a newer one

Reply:
I'll add a comment

Reply:
We cannot solve manually. See: https://github.com/kubernetes-sigs/kubebuilder/runs/5868585695?check_suite_focus=true#step:6:929

We generate all samples for each PR change and we ensure that the result of pr/repo is == what is done by the CLI. Otherwise, we never would have that properly updated. The point here is that by doing this change any project scaffold with the go/v2 plugin will have this dep added as well.  So, IMO we need to upgrade the go/v2 plugin too.

Reply:
I'm not sure I understand... I manually do that in the scaffolding - that's means I'm making sure that `go mod tidy` won't use a lower version.

Reply:
See:
- When we run make install -> we create a binary using the current code source 
- Then, when we run make generate -> we use the bin to execute an script which scaffolds all samples/examples under the testdata dir using the current version code

Then, when we push a pr we verify if the testdata dir is properly updated (by doing the same and using git diff)
So, if we go there and change what is inside of the testdata dir, the samples the tests will fail because the samples are NOT == what is the tool result.

Reply:
I think there's a misunderstanding here - what I meant is that I'm *specifically* generating it in the scaffolding template, and not letting `go mod tidy` to import this dependency.

Reply:
this file, is a `*.go` file...